src/deployment.js

```diff
@@ -109,63 +99,63 @@ class Deployment {
 
   // Poll the deployment endpoint for status
   async check() {
-    try {
-      const statusUrl =
-        this.deploymentInfo != null
-          ? this.deploymentInfo['status_url']
-          : `${this.githubApiUrl}/repos/${this.repositoryNwo}/pages/deployment/status/${this.buildVersion}`
-      let pageUrl = this.deploymentInfo != null ? this.deploymentInfo['page_url'] : ''
-      const previewUrl = this.deploymentInfo != null ? this.deploymentInfo['preview_url'] : ''
-      if (this.isPreview && previewUrl) {
-        pageUrl = previewUrl
-      }
-      core.setOutput('page_url', pageUrl)
-      const timeout = Number(core.getInput('timeout'))
-      const reportingInterval = Number(core.getInput('reporting_interval'))
-      const maxErrorCount = Number(core.getInput('error_count'))
-      var startTime = Date.now()
-      var errorCount = 0
+    const deploymentId = this.deploymentInfo?.id || this.buildVersion
+    const timeout = Number(core.getInput('timeout'))
+    const reportingInterval = Number(core.getInput('reporting_interval'))
+    const maxErrorCount = Number(core.getInput('error_count'))
+
+    let startTime = Date.now()
+    let errorCount = 0
 
-      // Time in milliseconds between two deployment status report when status errored, default 0.
-      var errorReportingInterval = 0
+    // Time in milliseconds between two deployment status report when status errored, default 0.
+    let errorReportingInterval = 0
 
+    try {
       /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
       while (true) {
         // Handle reporting interval
-        await new Promise(r => setTimeout(r, reportingInterval + errorReportingInterval))
+        await new Promise(resolve => setTimeout(resolve, reportingInterval + errorReportingInterval))
 
         // Check status
-        var res = await axios.get(statusUrl, {
-          headers: {
-            Authorization: `token ${this.githubToken}`
-          }
+        let res = await getPagesDeploymentStatus({
+          githubToken: this.githubToken,
+          deploymentId
         })
 
-        if (res.data.status == 'succeed') {
+        if (res.data.status === 'succeed') {
           core.info('Reported success!')
           core.setOutput('status', 'succeed')
+          if (this.deploymentInfo) {
```

Comment:
this is always truthy, thanks to line 57

Reply:
So long as the methods are call in the intended order. 😅  Less of a concern for the status check but I picture a future where we invoke `cancel()` potentially in a post-step if the workflow run fails or is canceled or something that may need to be able to operate without `this.deploymentInfo`. We're not there today, though, so your point is definitely valid. 👍🏻